draft-barnes-acme.md

```diff
@@ -222,6 +241,39 @@ To revoke a certificate, the client simply sends a revocation request, signed wi
 
 Note that while ACME is defined with enough flexibility to handle different types of identifiers in principle, the primary use case addressed by this document is the case where domain names are used as identifiers.  For example, all of the identifier validation challenges described in Section {identifier-validation-challenges} below address validation of domain names.  The use of ACME for other protocols will require further specification, in order to describe how these identifiers are encoded in the protocol, and what types of validation challenges the server might require.
 
+# Replay protection
+
+In order to prevent an attacker from replaying ACME messages against
+different ACME resources, or at a different time than they were sent,
+every authenticated ACME request body MUST include, in the Protected
+header, two Private Header Parameters {{RFC7515}}: "acmeURL" and
+"acmeNonce". The value of "acmeURL" MUST be equal to the absolute URL
```

Comment:
The "acmeURL" mechanism needs to be removed.  It is broken and unnecessary.

Signing the URL is unfortunately a bad match for HTTP.  It is very difficult for an HTTP server to reconstruct the URL used by the client to create a given request, especially in the bit-exact manner required for this mechanism to work.  If we keep this mechanism, it will be a debugging nightmare.

It's also unnecessary given the nonce mechanism below.  If you can't replay a message at all, then in particular you can't replay it against a different URI.